### DISCUSSION TITLE
RFC: Smart-hydration of different queries

### DISCUSSION THREAD

**Discussion by iduuck** [trpc/trpc#4328] [RFC / Ideas]
**Disclaimer:** I actually don't know an API for this as of now. But this just came into my mind while building something.

---

When building some sort of thing like you have a list, and then a detail page, when you click on the page.

When we say, there is a `blogPost.list` query on the list page. When navigating to the detail page of the blogpost, we could easily hydrate the `blogPost.byId` query with the information we have from the `blogPost.list` query and refetch in the background.

I know, that this is somehow something for react-query, but don't know, if they would even tackle this.

Here is a minimal illustration of this:

![Frame 2](https://user-images.githubusercontent.com/325241/236639157-91d39b00-6e66-4fb5-b2c2-322ee0e6b021.png)

**Comment by Nick-Lucas** [comment]
Yes this is definitely a react-query thing, and I'm pretty sure it's possible already!

You can setQueryData: https://tanstack.com/query/v4/docs/react/reference/QueryClient#queryclientsetquerydata

With tRPC it's also possible to get the query key to help with manipulation of this data. There is a code example here which may do what you need: https://trpc.io/docs/reactjs/getquerykey

**Reply by Nick-Lucas** [reply]
I think they have something more automated you can set up by the way, I thought it was called "Structural Sharing" but that seems to be on slightly different tracks at a glance.
